src/Tasks/Microsoft.NET.Build.Tasks/PublishRelativePathToPackPackagePath.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// 1. Publish PublishRelativePath is a relative <strong>file</strong> path. This need to convert to relative path.
+    /// 2. Due to DotnetTools package format, "tools/{TargetFramework}/any/" is needed in addition to the relative path.
+    /// </summary>
+    public sealed class PublishRelativePathToPackPackagePath : TaskBase
+    {
+        [Required]
+        public ITaskItem[] ResolvedFileToPublish { get; set; }
+
+        [Required]
+        public string PublishDir { get; set; }
+
+        [Required]
+        public string TargetFramework { get; set; }
+
+        [Output]
+        public ITaskItem[] ResolvedFileToPublishWithPackagePath { get; private set; }
+
+        private const char DirectorySeparatorChar = '\\';
+        private const char AltDirectorySeparatorChar = '/';
+
+        protected override void ExecuteCore()
+        {
+            var result = new List<TaskItem>();
+            foreach (ITaskItem r in ResolvedFileToPublish)
+            {
+                string relativePath = r.GetMetadata("RelativePath");
+                var fullpath = Path.GetFullPath(
+                    Path.Combine(PublishDir,
+                    relativePath));
+                var i = new TaskItem(fullpath);
+                i.SetMetadata("PackagePath", $"tools/{TargetFramework}/any/{GetDirectoryPathInRelativePath(relativePath)}");
+                result.Add(i);
+            }
+
+            ResolvedFileToPublishWithPackagePath = result.ToArray();
+        }
+
+        /// <summary>
+        /// Change "dir/file.exe" to "dir"
+        /// </summary>
+        internal static string GetDirectoryPathInRelativePath(string publishRelativePath)
```

Comment:
It is a simply task, but I find the most reliable way is to copy paste from corefx. Any suggestions?
See dotnet/corefx#4208